### DISCUSSION TITLE
[FEATURE REQUEST] Exempt hashtagged words from post character count

### DISCUSSION THREAD

**Discussion by glw907** [mastodon/mastodon#21526] [General]
Currently, all hashtags in a post count against the 500-character limit. This incents two bits of accessibility "bad behavior":

1. Shortened hashtags. For example, #akleg (6 characters) versus #AlaskaLegislature (18 characters). The second example is much better for folks using a screen reader.
2. Inline hashtag posting. Inserting hashtags at the bottom is better from an accessibility standpoint, but burns characters by forcing authors to repeat words.

One possible solution (aside from allowing longer posts) would be to exempt hashtagged items from the post character limit. (And you'd obviously need to limit the number of "free" hashtags in a post to prevent abuse of such a feature.)

And this is all said in the spirit of gratitude to the Mastodon dev team. Thanks for taking time to build software that makes the world a better place!

**Comment by kawazoe** [comment]
I came to propose something similar, but about CW and users. I came to a similar conclusion as you but for different reasons.

For me, I found it frustrating that some servers allow 1000+ char messages and others are much more limited. While it makes sense from a decentralized point of view, it also means that if a large conversation starts on one of those servers, I can't even participate because the user list in the reply would eat over 300 char from my post limit. Not only that, but user names are very large in Mastodon as it includes their server making the whole thing very verbose.

Add to this the fact that CW counts toward your character limit and you end up with a situation where people trying to do the right thing are penalized in their ability to discuss when compared to others.

Something I also just thought about while writing this post is that, if the message share the character limit with the CW, nothing prevents people from just typing their whole message in the CW and, by default, have it displayed in a larger font than other posts. I can see how easy this could be abused.

Even though smashing everything together in a single character limit is convenient from a technical perspective, I think it would be a much better experience if CW, headline users, trailing hashtags, and the message itself would be handled differently.

I propose:
- **60 char CW limit**. At 10 char of CW per ~50 char of text, I think this should be a good upper bound.
- **10 headline people limit**. This is around the limit of how many people you can have in a conversation before it gets too hard for our brain to follow what's happening (7 to 12 people).
  - This only applies to people in the headline. If you mention people throughout your message, then they should count toward your limit.
  - An alternative solution might be to not count the server part of the identifying toward the limit. This could help alleviate some of the unfairness as well.
- Instead of an independent trailing hashtag limit, I propose **a setting that pulls all hashtags into their own UI section** at the bottom of posts. This could fix the "repeating words" issues, since it would be done automatically for you so the duplication wouldn't count toward your character limit all while preventing SEO spamming.
- **400 char message limit**. With replies, CW and some hashtags out of the way, I'm happy with trimming the limit for messages a little to compensate.

**Reply by glw907** [reply]
FWIW, I think all of this makes sense from encouraging "good behavior", and this is especially timely as we have the ongoing Twitter influx. The pattern there is obviously quite different, and I think that we're seeing decreasing use of CWs in part because folks aren't willing to sacrifice the characters to include one. (I've honestly been guilty of exactly that behavior on longer posts, and I'm usually trying hard to respect the local conventions.)

And I believe that your suggestion to pulls all hashtags into their own UI section at the bottom of posts is a significant improvement on my original idea. With your suggestion, the UI would gently nudge folks to use hashtags in a more accessible way. And you'd probably want to impose a reasonable character limit for that section, just to prevent hashtag spamming.

**Reply by kawazoe** [reply]
@glw907 With my solution, there is no need for a separate limit for hashtags since it only list all of the hashtags from the post itself. In other words, a post like:

```
Hi @everyone, I really like #mastodon 's new #hashtaglist feature. #introduction
```

Could turn into:
```
Hi @everyone, I really like mastodon 's new hashtaglist feature.
----------------------
#mastodon #hashtaglist #introduction
```

Or, if we're trying to get clever with some word processing and trailing tag detection:
```
Hi @everyone, I really like mastodon's new hashtag list feature.
----------------------
#mastodon #hashtaglist #introduction
```

Since this is entirely front-end based, and the original post remains untouched, the post character limit would still apply. It can be done directly in the mobile app, or in the front-end portion of the mastodon server code, and could be tweaked with settings to enable extraction or word splitting independently. The best part is it doesn't depends on people to change their current behavior.

Either way, I think you are right that we need to change how the character limit is counted, just not for hashtags, since there is the option of doing some really neat stuff with them that we can't really do with mentions and CW.
